### DISCUSSION TITLE
Any way to continue a numbered list?

### DISCUSSION THREAD

**Discussion by paulmelis** [mkdocs/mkdocs#2838] [Q&A] [answered]
Is there a way to have multiple numbered lists, interspersed with paragraphs of text, with some way to have the automatic list item numbering continue at a specific value? Something like this as output:

```
1. First
2. Second
3. Third

A normal paragraph

4. Continue with item 4
5. Item 5
```

I've searched stackoverflow and found tips on using `{:start=4}` before the second list, and similar tricks, but nothing seems to work in mkdocs. I also can't find a plugin that could provide this. Any tips appreciated!

Edit: just realized that being able to turn off the automatic numbering for specific lists would be enough, but that doesn't seem to be possible either?

Edit 2: Using html lists is an option of course, but that would mean not being able to use markdown in the item itself. which I would like to avoid

**Comment by Andre601** [comment]
You need to use the `attr_list` extension which is part of the default Python Markdown and allows you to insert classes, fields, etc into sections of your markdown file.
In your case would `{ start=4 }` be probably used in the first list entry like this:

```markdown
4. Line { start=4 }
5. Line again
```

which should render into an ordered list like this:

```html
<ol start=4>
  <li>Line</li>
  <li>Line again</li>
</ol>
```

To add the `attr_list` extension, add this to your mkdocs.yml's `markdown-extensions` section:
```yaml
markdown-extensions:
- attr_list
```

[Documentation](https://python-markdown.github.io/extensions/attr_list/)

**Reply by paulmelis** [reply]
This does not seem to work for me, especially since what I read in https://github.com/Python-Markdown/markdown/issues/227 more or less says it's impossible with python-markdown (used by mkdocs) to attach an attribute to the list's `<ol>` tag, as that more-or-less doesn't really exist in markdown, as only the list items are markdown constructs. 

Testing directly with the `markdown` module seems to confirm this, the attribute is always attached to the `<li>` item, never the `<ol>`, unless I'm missing something

**Reply by paulmelis** [reply]
FYI, just submitted a pull request (https://github.com/Python-Markdown/markdown/pull/1252) to extend the attr_list syntax to allow setting list-wide attributes

**Comment by paulmelis** [comment]
The pull request mentioned above was not accepted in the Python-Markdown repository. 

Hence, I made a custom extension based on the existing attr_list extension. It can be found here: https://github.com/paulmelis/pm_attr_list

**Reply by valhuber** [reply]
I would dearly love to use this, but I have no idea where to "put the file so python can find it".  My doc project is here: https://github.com/ApiLogicServer/Docs

**Reply by renevanderark** [reply]
So, @paulmelis - seems like sane_lists fixed my problem, how did your issue differ from mine?

**Reply by paulmelis** [reply]
@valhuber Sorry for the late reply, didn't see your question. I just updated the repo, including some new setup instructions with pip.

@renevanderark I don't know. I remember trying sane_lists in the past, but then switched to mdx_truly_sane_lists (don't remember the reason).

**Comment by gledos** [comment]
```markdown
1.  First
2.  Second
3.  Third

    A normal paragraph

4.  Continue with item 4
5.  Item 5
```

---
1.  First
2.  Second
3.  Third

    A normal paragraph

4.  Continue with item 4
5.  Item 5
---

I think it is possible to indent the text of the paragraph.

**Reply by paulmelis** [reply]
But that’s not what I want. I need the option for multiple paragraphs per list item, but also paragraphs between list items, not indented.

**Reply by gledos** [reply]
I see, This really cannot be achieved using pure Markdown.

Thank you for your reply.

**Comment by renevanderark** [comment]
Is it not possible to pass pass the following option for the OListProcessor (LAZY_OL = False) through mkdocs config somehow?

https://github.com/Python-Markdown/markdown/blob/master/markdown/blockprocessors.py#L337

Setting it to False had exactly the desired numbering effect for me. 

The example here was built with the setting set to `True` hard-hacked into the markdown pip package installed on my local machine:
https://teaching-myself-godot.github.io/rewriting-zelia-tutorial/day-5/#setting-up-the-slimegd-script

**Reply by oprypin** [reply, marked as answer]
You can just add the following config

```yaml
markdown_extensions:
  - sane_lists
```

Setting `LAZY_OL` is an internal detail that is used by the `sane_lists` extension.

https://python-markdown.github.io/extensions/sane_lists/

https://github.com/Python-Markdown/markdown/blob/5d7f2559c31844f788ea41132f40ace7555518e6/markdown/extensions/sane_lists.py#L38

**Reply by renevanderark** [reply]
Ah thanks, apologies for pushing the issue!!
